vault_resource.go

```diff
@@ -189,3 +189,7 @@ func (r VaultResource) String() string {
 	}
 	return str
 }
+
+func (r VaultResource) ID() string {
+	return fmt.Sprintf("%s:%s", r.Resource, r.Path)
```

Comment:
I wonder should we just refer to everything by path. We don't really mind how sidekick is going to parse the info.
We simply know that vault has a unique path, and that path has a last issued TTL